### PR TITLE
Added Type definitions for Woosmap Map JS API 1.4

### PR DESCRIPTION
**Pull request by gaelsimon**
Initialize Type definitions for non-npm Global Library `woosmap.map` 

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/woosmap.map/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+      "index.d.ts",
+        "woosmap.map-tests.ts"
+    ]
```
Comment (peterblazejewicz) [inline types/woosmap.map/tsconfig.json:23]:
```suggestion
    "files": ["index.d.ts", "woosmap.map-tests.ts"]
```

**Comment by peterblazejewicz** [comment]
- every of this person listed is required to give explicit ack in order to be listed here, as DT is sending notifications to module owners. Person can be just named via @credit to avoid notification (giving credit is ok and does not require ACK)
```
// Definitions by: Emmanuel Peralta <https://github.com/manz>
//                 Samuel Mendes <https://github.com/smendes>
//                 Gael Simon <https://github.com/gaelsimon>
```
- the code is autogenerated, but I would advise to run DT formatting as described in DT README:
```bash
npm run prettier -- --write .\types\woosmap.map\**\*.ts
```

File: types/woosmap.map/tslint.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "ban-types": false,
+        "max-line-length": false,
+        "no-mergeable-namespace": false,
+        "no-unnecessary-class": false,
+        "no-unnecessary-qualifier": false
```
Comment (sandersn) [inline types/woosmap.map/tslint.json:8]: Why is `no-unnecessary-class` failing?
It's difficult with generated code, but ideally no lint rules would have to be disabled.

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/woosmap.map/tslint.json (hunk 2 above)
Comment (gaelsimon) [reply types/woosmap.map/tslint.json]: Thanks for your feedback!
We fixed the generated type defs to be able to remove additional lint rules.
The rules `no-unnecessary-qualifier` will be hard to remove as some interfaces have common names but under different sub namespaces. Same for `no-mergeable-namespace` which will be hard to fix in our generation tool.

**Review by manz — APPROVED** [review]
LGTM 👍

**Review by smendes — APPROVED** [review]

**Pull request merged**